src/main/resources/META-INF/rewrite/java-version-25.yml

```diff
@@ -0,0 +1,48 @@
+#
+# Copyright 2024 the original author or authors.
+# <p>
+# Licensed under the Moderne Source Available License (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# <p>
+# https://docs.moderne.io/licensing/moderne-source-available-license
+# <p>
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.migrate.UpgradeToJava25
+displayName: Migrate to Java 25
+description: >-
+  This recipe will apply changes commonly needed when migrating to Java 25. This recipe will also replace deprecated API
+  with equivalents when there is a clear migration strategy. Build files will also be updated to use Java 25 as the
+  target/source and plugins will be also be upgraded to versions that are compatible with Java 25.
+tags:
+  - java25
+recipeList:
+  - org.openrewrite.java.migrate.RemoveSecurityManagerAPI
+
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.migrate.RemoveSecurityManagerAPI
+displayName: Remove Security Manager API
+description: The Security Manager API is unsupported in Java 24. This recipe will remove the API and any usages of it.
+tags:
+  - java25
+preconditions:
+  - org.openrewrite.java.migrate.search.FindJavaVersion:
+      version: 24
+recipeList:
+  - org.openrewrite.java.RemoveMethodInvocations:
+      methodPattern: java.security.AccessController#checkPermission(..)
+  - org.openrewrite.java.RemoveMethodInvocations:
+      methodPattern: java.security.Policy#setPolicy(..)
+  - org.openrewrite.java.RemoveMethodInvocations:
+      methodPattern: java.lang.SecurityManager#check*(..)
+  - org.openrewrite.staticanalysis.UnnecessaryCatch
+  - org.openrewrite.staticanalysis.UnnecessaryThrows
```

Comment:
Hmm; `UnnecessaryCatch` and `UnnecessaryThrows` will make changes anywhere, not just where it relates to the changes stemming from the removed methods. That might surprise folks that run a Java upgrade to see these changes, especially when they were not using anything related to `java.security` or `java.lang.SecurityManager`.  Perhaps we an add a precondition to `RemoveSecurityManagerAPI` at the very least, or move that into an imperative recipe that only runs the `Unnecessary` recipes when there are changes, and only on affected files.

Reply:
Yes a precondition for the 3 types in use make sence

Reply:
Does it make sence to make it 3 refcipes?